docs/design/proposals/codegen-refactor/model_cmd_2.md

```diff
@@ -0,0 +1,143 @@
+* `shape` | `aws-sdk` | `sdk-shape` | `sdk`: the original operations, models, errors, structs for a given AWS service. sourced from *aws-sdk*, ex: [aws-sdk-go s3](https://github.com/aws/aws-sdk-go/blob/main/models/apis/s3/2006-03-01/api-2.json#L1)
+* `reconciliation`: logic involving access to or relations between `resource`, `shape`, `ackgenconfig`, and `aws-sdk`
+* `ackmodel`: the [code](https://github.com/aws-controllers-k8s/code-generator/blob/main/pkg/model/model.go#L36) representation of ACK's view of the world; the source of truth for `aws-sdk`, `ackgenconfig`, and `reconciliation`
+
+## Problem
+There are issues that are starting to degrade the code generation `pipeline` experience and will compound as *code-generator* continues to grow:
+* Duplicate work between `api` and `controller` commands; both generate `ackmodel`
+* Overloaded functions such as [SetResource](https://github.com/aws-controllers-k8s/code-generator/blob/main/pkg/generate/code/set_resource.go#L74); responsible for a lot more than generating Go code
+* Confusing CX. There's a not-so-clear dependency on a folder in the $SERVICE-controller via [getLatestAPIVersion](https://github.com/aws-controllers-k8s/code-generator/blob/26e5da2e7656bb836ee438c05df14f2adc50197d/cmd/ack-generate/command/common.go#L271). This goes against the flow of the `pipeline` and creates confusion as there is a `--version` flag for the `apis` command (the above function takes precedence).
+* Long dev loop; after making changes to either *generator.yaml* or logic in *code-generator* users need to either manually verify files or run e2e tests-- both of which are time consuming today
+* How *code-generator* reconciles `ackgenconfig` with `aws-sdk` to create `ackmodel` is inconsistent (and some duplicated work). For example, overrides such as `custom_shapes` [edit aws-sdk directly](https://github.com/aws-controllers-k8s/code-generator/blob/main/pkg/sdk/custom_shapes.go#L62) while others such as `ignored_operations` [set operation values to nil](https://github.com/aws-controllers-k8s/code-generator/blob/26e5da2e7656bb836ee438c05df14f2adc50197d/pkg/model/model.go#L295) in memory leaving the actual operations in `aws-sdk` untouched.
+
+Current Pipeline
+---
+
+![current-pipeline](./images/current_pipeline.png)
+* Displayed are the important and mostly duplicated calls between the `apis` and `controller` commands
+* The cache is only used to store the `aws-sdk` repo
+* The red line denotes a dependency on $SERVICE-controller (output directory) repo that should be removed
+* Only `GetCRDs` is duplicated between the commands when creating the model
+
+
+Current Dev Loop
+---
+
+![current-dev-loop](./images/current_dev_loop.png)
+* Validating the result of a change in *code-generator* and/or *generator.yaml* can be a lengthy process
```

Comment:
I don't understand the "save previous `apis/` `pkg/`" in this diagram? Is this referring to verifying the git diff after generating code?